lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,344 @@
+import boto3
+import json
+import logging
+import os
+import psycopg2
+import requests
+import sys
+from datetime import datetime
+
+# Constants
+AWS_REGION = "us-gov-west-1"
+START_TYPES = ('START', 'BEGIN', 'RESTART', 'OPTIN', 'OPT-IN')
+STOP_TYPES = ('STOP', 'OPTOUT', 'OPT-OUT')
+HELP_TYPES = ('HELP',)
+START_TEXT = 'Message service resumed, reply "STOP" to stop receiving messages.'
+STOP_TEXT = 'Message service stopped, reply "START" to start receiving messages.'
+HELP_TEXT = 'Some help text'
+INBOUND_NUMBERS_QUERY = """SELECT number, service_id, url_endpoint, self_managed FROM inbound_numbers;"""
+
+# Validation set.  Valid event data must have these attributes.
+EXPECTED_PINPOINT_FIELDS = frozenset(('originationNumber', 'destinationNumber', 'messageBody'))
+
+# Environment variables set by the lambda infrastructure
+AWS_PINPOINT_APP_ID = os.getenv("AWS_PINPOINT_APP_ID")
+DEAD_LETTER_SQS_URL = os.getenv("DEAD_LETTER_SQS_URL")
+LOG_LEVEL = os.getenv('LOG_LEVEL')
+RETRY_SQS_URL = os.getenv('RETRY_SQS_URL')
+
+TIMEOUT = os.getenv('TIMEOUT')
+TIMEOUT = tuple(TIMEOUT) if isinstance(TIMEOUT, list) else int(TIMEOUT)
+
+logger = logging.getLogger('TwoWaySMSv2')
+logger.setLevel(logging.getLevelName(LOG_LEVEL))
+
+################################################################################################
+# Get the database URI from SSM Parameter Store.
+################################################################################################
```

Comment:
Stopping the review here until the removed methods is put back or sufficient justification is submit.

Reply:
Making it a method call from the handler is something we thought we needed to do for reasons related to user permissions (lambda environment setup user vs. handler user), but that turned out to be unnecessary.  I refactored it to flow like my VA Profile integration lambda.  The db query should happen once, when the module loads.

Please also note that my re-use of the NOTIFY_ENVIRONMENT, which I created for Profile integration, allows me to preempt boto3 calls without having to worry about mocking anything in unit tests.

Reply:
I understand the `NOTIFY_ENVIRONMENT` addition and have no issues with that one. Going to move this conversation to zoom/slack.